src/server/engine_shard_set.cc

```diff
@@ -70,6 +79,66 @@ ShardMemUsage ReadShardMemUsage(float wasted_ratio) {
   return usage;
 }
 
+// TODO: document this class.
+// Thread safe.
+class RoundRobinSharder {
+ public:
+  void InitThreadLocal() {
+    round_robin_prefix_ = absl::GetFlag(FLAGS_round_robin_prefix);
+
+    if (IsEnabled()) {
+      // 500k entries will consume 1mb per thread, and will allow 100 keys with < 1% collision
+      // probability. Since this has a considerable footprint, we only allocate when enabled.
+      constexpr size_t kRoundRobinSize = 500'000;
```

Comment:
I suggest using a prime number in the vicinity of 500K instead of 500K.
with non-prime numbers the utilization of the table will be much worse